lib/iris/_lazy_data.py

```diff
@@ -158,3 +158,28 @@ def multidim_lazy_stack(stack):
         result = da.stack([multidim_lazy_stack(subarray)
                            for subarray in stack])
     return result
+
+
+def wrap_lazy_elementwise(lazy_array, elementwise_op):
+    """
+    Apply a (numpy-style) elementwise array operation to a lazy array.
+
+    Elementwise means that it performs a independent calculation at each point
+    of the input, producing a result array of the same shape.
+
+    Args:
+
+    * lazy_array:
+        The lazy array object to operate on.
+    * elementwise_op:
+        The elementwise (numpy) array operation to apply.
+
+    """
+    # This is just an Iris wrapper for the Dask operation.
+    # For now, we support only a single argument array, and assume that the
+    # output dtype is the same as the input.  This scope is intentionally
```

Comment:
I much dithered about that, also whether we should support different numbers and shapes of result, as map_blocks does.
I guess that doing exactly what map+blocks does at least means it can't get it wrong when you don't specify, but it's a slightly tricky behaviour and I'm slightly troubled by the copycat approach.

I think I'd prefer a *required* dtype arg.  How does that sound to you @dkillick ?

Reply:
I'd be in favour of making it required, to avoid Dask doing a test operation. I think you;re going to need to add it anyway to fix the bug I mentioned in my other comment.